### DISCUSSION TITLE
Chain vdb_view and vdb_render

### DISCUSSION THREAD

**Discussion by chrisflesher** [AcademySoftwareFoundation/openvdb#1715] [Q&A] [answered]
I was wondering if there is an existing way to use the `vdb_view` tool to determine good camera pose for the `vdb_render` tool? For example, is there a way to print the current `vdb_view` camera pose? Or any other suggestions?

**Comment by danrbailey** [comment, marked as answer]
Hi Chris, it's a great idea, we haven't implemented anything like this, but it doesn't sound too complicated if you'd like to have a go at this yourself? We are not sure if the cameras are the same however. One other thing to mention is that there is a flag to vdb_render which allows you to look at a specific xyz location which can also be quite convenient:
https://github.com/AcademySoftwareFoundation/openvdb/blob/master/openvdb_cmd/vdb_render/main.cc#L188

**Reply by chrisflesher** [reply]
Ok, I am wondering if I tried to make a PR what you think would be acceptable? I guess my first proposal would be to have `vdb_view` write an image to disk when you hit a key with an incrementing number (e.g. `snapshot_0.png`, `snapshot_1.png`). Or maybe a second proposal would be to print the current camera pose when you hit a key. Does one of these sound acceptable or maybe you have more ideas?

**Reply by chrisflesher** [reply]
Or are you suggesting I should create a new application for my own purposes and not necessarily attempt to alter the existing tool? I don't really mind either way, am just trying to understand what you're suggesting exactly.

**Reply by danrbailey** [reply]
I was suggesting you make a PR for this and we can try and include it? My initial thought was to just have a flag to `vdb_view` where it would output the camera position whenever you moved the camera, but the idea of pressing a key to output the camera position to the shell sounds good to me too.

**Reply by chrisflesher** [reply]
Ok, will try it, thanks!
